### ISSUE TITLE
Detect LLVM 3.5 prerelease from Ubuntu 14.04 in setup.py and warn users

### ISSUE THREAD

**Issue by seibert** [numba/llvmlite#30]
The LLVM 3.5 prerelease that ships with Ubuntu 14.04 is going to cause confusion for years to come.  We should detect this version and warn users that the pre-release is not supported.

**Comment by seibert** [comment]
Here is the Ubuntu bug tracking this issue:

https://bugs.launchpad.net/ubuntu/+source/llvm-toolchain-3.5/+bug/1372062

LLVM provides a package repository we should point users towards:

http://llvm.org/apt/

**Comment by sontek** [comment]
+1, I haven't solved the problem yet but I've been searching around for awhile because I have llvm-3.5 on ubuntu 14.04 and getting a bunch of errors.   I think documentation on getting this to work would be important.

**Comment by arve0** [comment]
Is this still an issue? This works:

```sh
sudo apt-get install llvm-3.5-dev zlib1g-dev libedit-dev
LLVM_CONFIG=/usr/lib/llvm-3.5/bin/llvm-config pip install llvmlite
```

```sh
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.2 LTS
Release:	14.04
Codename:	trusty
```
```sh
$ apt-cache show llvm-3.5-dev
Package: llvm-3.5-dev
Priority: optional
Section: devel
Installed-Size: 79146
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
Architecture: amd64
Source: llvm-toolchain-snapshot
Version: 1:3.5-4ubuntu2~trusty2
Replaces: llvm (<< 2.2-3)
Depends: libc6 (>= 2.14), libllvm3.5, libstdc++6 (>= 4.6), libffi-dev (>= 3.0.9), llvm-3.5 (= 1:3.5-4ubuntu2~trusty2), libtinfo-dev
Filename: pool/main/l/llvm-toolchain-snapshot/llvm-3.5-dev_3.5-4ubuntu2~trusty2_amd64.deb
Size: 11192520
MD5sum: 3da2a5b449072b423ee94fce8b16c9ba
SHA1: aeb91ae4430e336267bcf9ce3d8f9a35e058a738
SHA256: 2189dfe097883ad5fef622f174c8663986145663c99c2d3282de3c51e360d1da
Description-en: Modular compiler and toolchain technologies, libraries and headers
 LLVM is a collection of libraries and tools that make it easy to build
 compilers, optimizers, just-in-time code generators, and many other
 compiler-related programs.
 .
 LLVM uses a single, language-independent virtual instruction set both
 as an offline code representation (to communicate code between
 compiler phases and to run-time systems) and as the compiler internal
 representation (to analyze and transform programs). This persistent
 code representation allows a common set of sophisticated compiler
 techniques to be applied at compile-time, link-time, install-time,
 run-time, or "idle-time" (between program runs).
 .
 This package provides the libraries and headers to develop applications
 using llvm.
Description-md5: 29f654aa391c01f742a2023c872d7a49
Homepage: http://www.llvm.org/
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Origin: Ubuntu
Supported: 9m

Package: llvm-3.5-dev
Priority: optional
Section: universe/devel
Installed-Size: 72094
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debiUploaders: Sylvestre Ledru <sylvestre@debian.org>
Architecture: amd64
Source: llvm-toolchain-snapshot
Version: 1:3.5~svn201651-1ubuntu1
Replaces: llvm (<< 2.2-3)
Depends: libc6 (>= 2.14), libllvm3.5, libstdc++6 (>= 4.6), libffi-dev (>= 3.0.9), llvm-3.5 (= 1:3.5~svn201651-1ubuntu1), libtinfo-dev
Filename: pool/universe/l/llvm-toolchain-snapshot/llvm-3.5-dev_3.5~svn201651-1ubuntu1_amd64.deb
Size: 10310540
MD5sum: 4b6e438ab0987a285aafa0dae6700238
SHA1: 413b148f26a476eb2eee51b7956f444a617aec85
SHA256: eee546e53771bcc7d6626eb5af3863cb43e58d9ee195f30250f1d7456f5d759e
Description-en: Modular compiler and toolchain technologies, libraries and headers
 LLVM is a collection of libraries and tools that make it easy to build
 compilers, optimizers, just-in-time code generators, and many other
 compiler-related programs.
 .
 LLVM uses a single, language-independent virtual instruction set both
 as an offline code representation (to communicate code between
 compiler phases and to run-time systems) and as the compiler internal
 representation (to analyze and transform programs). This persistent
 code representation allows a common set of sophisticated compiler
 techniques to be applied at compile-time, link-time, install-time,
 run-time, or "idle-time" (between program runs).
 .
 This package provides the libraries and headers to develop applications
 using llvm.
Description-md5: 29f654aa391c01f742a2023c872d7a49
Homepage: http://www.llvm.org/
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Origin: Ubuntu
```

**Comment by pitrou** [comment]
> Is this still an issue? This works:

Really? Well, interesting. Regardless, we're switching llvmlite to LLVM 3.6 now, so this is obsolete (unless Ubuntu comes with another broken package :-)).
